### DISCUSSION TITLE
Feature requests: Queue until webui is available & Wake on Lan

### DISCUSSION THREAD

**Discussion by thulle** [Kilvoctu/aiyabot#145] [Ideas]
Thanks for the bot! Real fun to have to bounce stupid ideas for images with friends.

We run the bot on our shared NAS and send the API request to a webui on a workstation.
Some things that would improve our use case is:

 - The ability to toggle the bot to not timeout connections to the webui, but continue trying and just add new draw requests to the queue. This way draw requests won't have to be redone later if the workstation isn't available and we can add image ideas to the bot whenever they come to mind.
 - This should preferably also not abort on a 503 Service Unavailable or 504 Gateway Timeout, this way we can add a http load balancer to distribute the job to whoever has a webui running currently.
 - Another nice feature would be the possibility to configure an URL that aiyabot can send a request to if the request times out, this way we could add an endpoint to wake a workstation up via Wake on Lan, let it process the job, and then go back to sleep. We can probably trigger this one from the load balancer though, so not as necessary a feature, and a bit more work to add compared to looping on the API-request attempt.

**Comment by Kilvoctu** [comment]
Heya, I'm not a network person so I only understood like half of that lol 😅. I went ahead and [added it to the list of projects](https://github.com/users/Kilvoctu/projects/14) to look into it later.  
AIYA has a makeshift way to wait on Web UI on startup, which only sorta works. I've always wanted to go back and improve it. In doing that, I can add try to add that behavior to queuing, too.  

Currently, I was working on [the GUI project](https://github.com/users/Kilvoctu/projects/1) when my PC setup suffered hardware failure, so I'm a bit bummed out at the moment to do much developing. Once I get back to it, I'll see if I can implement the first request; sometimes these things are easy when given a try. 👍
